### DISCUSSION TITLE
Sync metadata from iPhone to Mac

### DISCUSSION THREAD

**Discussion by RhetTbull** [RhetTbull/osxphotos#1480] [General]
In response to a question on Reddit, I documented the following steps to sync metadata (in this case favorites and albums, though you can also sync title, keywords, and description) from the iPhone to the Mac. This is documented in the help for the [osxphotos sync](https://rhettbull.github.io/osxphotos/cli.html#osxphotos-sync) command but the following provides more detailed steps:

To sync metadata from iPhone to Photos on the Mac

- Install [iMazing](https://imazing.com/)
- Plug the phone in to the Mac via a USB cable. If prompted, select "Trust this device"
- Start iMazing
- Expand the iPhone icon in the left panel in the iMazing window
- Scroll down and select`File System` from the list in the left panel under the iPhone icon
- Open the `Media` folder
- Open the `PhotoData` folder
- Find the file `Photos.sqlite` and copy it to your Mac by dragging it from the iMazing window. There may also be files named `Photos.sqlite-shm` and `Photos.sqlite-wal` in the same folder. Copy these as well.
- Open the Terminal and run the following command:

`osxphotos sync --import /path/to/Photos.sqlite --set favorite,albums --verbose`

replacing `/path/to/Photos.sqlite` with the path to wherever you copied the `Photos.sqlite` file.

If you want to do a dry run and see what will be synced without actually changing the photos library, add `--dry-run`:

`osxphotos sync --import /path/to/Photos.sqlite --set favorite,albums --verbose --dry-run`

**Comment by Serhansolo** [comment]
Hi, man. I wonder if this will solve my problem. I was hesitant to create a new topic, but the problem I want to solve concerns metadata.

Furthermore, I have owned an iPhone since 2018, and gradually I have adopted several devices. Apple does an impressive job of making outstanding devices. I just hate the fact that the biggest catch-22 with iCloud is trapping you into their never-ending update cycle for photos. Since I have never opted to use photo syncing, I have been using the Photos app on my MacBook to create a Photos Library on an external HDD. Ever since, I have been particularly careful about having a good strategy for culling and keeping things organized. I always keep a "handful" of favorites of my best moments on my iPhone; I also edit all the pictures I keep as favorites.

For some reason, it escaped my attention that all the photos I imported into the Photos Library on my MacBook only kept the edits and the Live Presentation of the image, but not the little "hearted" favorite metadata. This kind of made me furious, I have to admit.

Anyway, long story short. I just want to keep my favorites synced in this Photos Library that I have been managing for so long. Can your tool be of any assistance?

P.S.
I you would suggest that I should make a new topic for this I will gladly move this question :)

**Reply by RhetTbull** [reply]
Edit: Sorry, I was responding on mobile and didn't realize this was a thread. Short answer: yes, following the directions above (and pasted below) should allow you to sync your favorites. This does require jumping through some hoops to get the iPhone Photos database off the phone then running a command in the Terminal. 

This is possible using  [osxphotos sync](https://rhettbull.github.io/osxphotos/cli.html#osxphotos-sync) command but it's not easy as you've got to first copy the Photos database off the phone. Here's the basic instructions:

To sync metadata from iPhone to Photos on the Mac

- Install [iMazing](https://imazing.com/) (there may be other options but iMazing is the only I've tested)
- Plug the phone in to the Mac via a USB cable. If prompted, select "Trust this device"
- Start iMazing
- Expand the iPhone icon in the left panel in the iMazing window
- Scroll down and select`File System` from the list in the left panel under the iPhone icon
- Open the `Media` folder
- Open the `PhotoData` folder
- Find the file `Photos.sqlite` and copy it to your Mac by dragging it from the iMazing window. There may also be files named `Photos.sqlite-shm` and `Photos.sqlite-wal` in the same folder. Copy these as well.
- Open the Terminal and run the following command:

`osxphotos sync --import /path/to/Photos.sqlite --set favorite,albums --verbose`

replacing `/path/to/Photos.sqlite` with the path to wherever you copied the `Photos.sqlite` file.

If you want to do a dry run and see what will be synced without actually changing the photos library, add `--dry-run`:

`osxphotos sync --import /path/to/Photos.sqlite --set favorite,albums --verbose --dry-run`

This syncs favorites and albums. To sync only favorites, replace `favorites,albums` with `favorites`. See `osxphotos help sync` for more information on the sync command.
